src/pal/src/pal_load_file.c

```diff
@@ -1,24 +1,28 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include "pal_log.h"
+#include "pal_load_file.h"
 
-char *pal_load_file_to_string(const char *filename) {
+void pal_load_file(const char *filename, load_file_t *load_file) {
     FILE *fp = fopen(filename, "rb");
 
     if (fp == NULL) {
         PAL_WARN("Warning: Failed to open file: %s", filename);
-        return NULL;
+        return;
     }
     fseek(fp, 0, SEEK_END);
     long fsize = ftell(fp);
+
     fseek(fp, 0, SEEK_SET);
-    char *file_buffer = malloc(fsize + 1);
-    if (file_buffer == NULL) {
+    load_file->buffer = malloc(fsize + 1);
+    if (load_file->buffer == NULL) {
         PAL_WARN("Warning: Failed to malloc buffer for file: %s", filename);
-        return NULL;
+        return;
     }
-    fread(file_buffer, 1, fsize, fp);
-    file_buffer[fsize] = '\0';
+    fread(load_file->buffer, 1, fsize, fp);
+    load_file->buffer[fsize] = '\0';
+    load_file->size = fsize;
```

Comment:
load_file->size = fsize + 1;
size contains the null terminator.

Reply:
Done, nice suggestion!